hack/bump-k8s.sh

```diff
@@ -34,6 +34,6 @@ BRANCH=${BRANCH:-feature-logical-clusters-1.23}
 current_version="$( GOPROXY=direct go mod edit -json | jq '.Replace[] | select(.Old.Path=="k8s.io/kubernetes") | .New.Version' --raw-output )"
 
 # equivalent to go mod edit -replace
-sed -i'' -e "s|${current_version}|${BRANCH}|g" -E -e "s,=> github.com/[^/]+/[a-zA-Z0-9_-]+,=> github.com/${GITHUB_USER}/${GITHUB_REPO},g" go.mod
+sed -i '' -e "s|${current_version}|${BRANCH}|g" -E -e "s,=> github.com/[^/]+/[a-zA-Z0-9_-]+,=> github.com/${GITHUB_USER}/${GITHUB_REPO},g" go.mod
```

Comment:
I thought this was supposed to work for both of us? Damn it. Can we do a conditional here based on the platform lol?

Reply:
```suggestion
sed -i -e "s|${current_version}|${BRANCH}|g" -E -e "s,=> github.com/[^/]+/[a-zA-Z0-9_-]+,=> github.com/${GITHUB_USER}/${GITHUB_REPO},g" go.mod
```
Why do you think the `-i''` does not work? Seems to work here as long as `-e` is following, and you can leave out the`''`, both with sed and gsed.

Reply:
Because I'm on Linux and added the space to make it work for me last time I used the script :)

Reply:
What am I missing? Works like a charm.

<img width="561" alt="Bildschirmfoto 2022-07-28 um 09 46 12" src="https://user-images.githubusercontent.com/730123/181450453-b6ec50ae-ba6b-40ad-a472-a1836e6e0400.png">